### PR TITLE
Firefly 1533 runid jobname

### PR DESCRIPTION
**Pull request by loitly**
Ticket: https://jira.ipac.caltech.edu/browse/FIREFLY-1533

- attaching a RUNID to the submission when TAP service supports it.
- save information local for display
- filling it in with the user-specified value
- displaying it in the job-status dialog
- displaying it in the table-info dialog

Test: https://fireflydev.ipac.caltech.edu/firefly-1533-runid-jobname/firefly/
- do a TAP search
- [optional] enter a custom title.  (bottom of form next to Search and Row Limit)
- while in progress, click on 'i' icon to see Job Information, then 'Show' to see UWS Job info.
- once table results returned, click on 'i' table info to see 'Job Info', then 'Show' to see UWS Job info.

Notes:
- most of the predefined TAP services do not properly support this.
- RUNID is limited to 64 chars.  If more than 64, truncate then add '...' to indicate it was truncated.
- when it is not supported, our UWS Job popup may show a different value than if you click the link that takes you to their job info url.

### CODE REVIEW

File: src/firefly/js/ui/tap/TapKnownServices.js [hunk 1]
```diff
@@ -28,11 +28,11 @@ function makeServices() {
         tapEntry('MAST Images', 'https://vao.stsci.edu/CAOMTAP/TapService.aspx'),
         tapEntry('CADC', 'https://ws.cadc-ccda.hia-iha.nrc-cnrc.gc.ca/argus/'),
         // CDS???
-        tapEntry('VizieR (CDS)', 'http://tapvizier.u-strasbg.fr/TAPVizieR/tap/'),
+        tapEntry('VizieR (CDS)', 'https://tapvizier.u-strasbg.fr/TAPVizieR/tap/'),
         tapEntry('Simbad (CDS)', 'https://simbad.u-strasbg.fr/simbad/sim-tap'),
         // more ESA??
         tapEntry('Gaia', 'https://gea.esac.esa.int/tap-server/tap',undefined,undefined,undefined, undefined, gaiaExamples()),
-        tapEntry('GAVO', 'http://dc.g-vo.org/tap'),
+        tapEntry('GAVO', 'https://dc.g-vo.org/tap'),
```
Comment (robyww) [inline src/firefly/js/ui/tap/TapKnownServices.js:35]: good, Thanks for catching that!

File: src/firefly/java/edu/caltech/ipac/firefly/server/query/AsyncTapQuery.java [hunk 2]
```diff
@@ -29,19 +31,38 @@ public class AsyncTapQuery extends UwsJobProcessor {
...
+    private static final List<String> SVC_RUNID_NOT_SUPPORTED = Arrays.asList(
+            "https://irsa.ipac.caltech.edu/TAP",                // Return exception, BAD_REQUEST: RUNID not implemented
+            "https://exoplanetarchive.ipac.caltech.edu/TAP/",   // Bad implementation. it replaces it with its own identifier, e.g. 109294
+            "https://koa.ipac.caltech.edu/TAP/",                // Bad implementation. it replaces it with its own identifier, e.g. 109294
+            "https://heasarc.gsfc.nasa.gov/xamin/vo/tap",       // Accepted the parameter, but did not return its value
+            "https://vao.stsci.edu/CAOMTAP/TapService.aspx",    // Accepted the parameter, but did not return its value
+            "https://gea.esac.esa.int/tap-server/tap",          // Accepted the parameter, but did not return its value
+            "https://dc.g-vo.org/tap",                          // Accepted the parameter, but did not return its value
+            "https://archives.esac.esa.int/hsa/whsa-tap-server/tap"     // Accepted the parameter, but did not return its value
+    );
```
Comment (robyww) [inline src/firefly/java/edu/caltech/ipac/firefly/server/query/AsyncTapQuery.java:44]: Maybe only exceptions should only be on this list and not ones that ignore the parameter. If they fix it then we will still not send the runid.

**Review by robyww — APPROVED** [review]
It works. It appears that I could only test with CADC.

All the code looks good but I am concerned that the black list if too aggressive.

File: src/firefly/java/edu/caltech/ipac/firefly/server/query/AsyncTapQuery.java (hunk 2 above)
Comment (loitly) [reply src/firefly/java/edu/caltech/ipac/firefly/server/query/AsyncTapQuery.java:44]: Sure.  I can limit the list to only IRSA.  I'll keep the rest as comments so we know what to expect from these services.

**Pull request merged**